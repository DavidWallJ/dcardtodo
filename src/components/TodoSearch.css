form {
	background-color: #f9f9f9;
}

.showCompletedCheckbox {
	text-align: left;
	padding-left: 1px;
}

.todoSearchForm input[type='checkbox'] {
	margin-top: 2px;
}
