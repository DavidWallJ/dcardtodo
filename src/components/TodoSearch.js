import React, { Component } from 'react';
import { Button, FormGroup, FormControl, Checkbox } from 'react-bootstrap';

class TodoSearch extends Component {
	handleSearch = () => {
		const searchInput = this.searchInput.value;
		// passed down from parent component
		this.props.onSearch(searchInput);
	};

	handleCheckbox = e => {
		const showCompleted = e.target.checked;
		// passed down from parent component
		this.props.onCheckbox(showCompleted);
	};

	render() {
		return (
			<form>
				<FormGroup>
					<FormControl
						type="search"
						className="todoSearchInput"
						placeholder="Search"
						inputRef={input => (this.searchInput = input)}
						onChange={this.handleSearch}
					/>
					<Checkbox checked={this.props.checked} onChange={this.handleCheckbox}>
						Show completed todos
					</Checkbox>
				</FormGroup>
			</form>
		);
	}
}

export default TodoSearch;
