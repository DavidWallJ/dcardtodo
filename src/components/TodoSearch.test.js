import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TodoSearch from './TodoSearch';

describe(TodoSearch, () => {
	const searchText = 'Buy Milk';
	const mockOnSearch = jest.fn();
	const component = shallow(<TodoSearch onSearch={mockOnSearch} />);

	it('should exist', () => {
		const component = renderer.create(<TodoSearch />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('contains the form', () => {
		expect(component.find('form')).toHaveLength(1);
		expect(component.find('Checkbox')).toHaveLength(1);
		expect(component.find('.todoSearchInput')).toHaveLength(1);
	});
});
