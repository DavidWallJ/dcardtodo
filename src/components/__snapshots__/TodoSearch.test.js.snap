// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class TodoSearch extends _react.Component {constructor(...args) {var _temp;return _temp = super(...args), this.
		handleSearch = () => {
			const searchInput = this.searchInput.value;
			// passed down from parent component
			this.props.onSearch(searchInput);
		}, this.

		handleCheckbox = e => {
			const showCompleted = e.target.checked;
			// passed down from parent component
			this.props.onCheckbox(showCompleted);
		}, _temp;}

	render() {
		return (
			_react2.default.createElement('form', { __source: { fileName: _jsxFileName, lineNumber: 19 }, __self: this },
				_react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: this },
					_react2.default.createElement(_reactBootstrap.FormControl, {
						type: 'search',
						className: 'todoSearchInput',
						placeholder: 'Search',
						inputRef: input => this.searchInput = input,
						onChange: this.handleSearch, __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }),

					_react2.default.createElement(_reactBootstrap.Checkbox, { checked: this.props.checked, onChange: this.handleCheckbox, __source: { fileName: _jsxFileName, lineNumber: 28 }, __self: this }, 'Show completed todos'))));





	}} should exist 1`] = `
<form>
  <div
    className="form-group"
  >
    <input
      className="todoSearchInput form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="Search"
      type="search"
    />
    <div
      className="checkbox"
      style={undefined}
    >
      <label
        title=""
      >
        <input
          checked={undefined}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        Show completed todos
      </label>
    </div>
  </div>
</form>
`;
